src/main/java/com/digitalpebble/storm/crawler/bolt/SiteMapParserBolt.java

```diff
@@ -0,0 +1,178 @@
+        for (Values ol : outlinks) {
+            collector.emit(Constants.StatusStreamName, new Values(ol));
+        }
+
+        // marking the main URL as successfully fetched
+        collector.emit(Constants.StatusStreamName, new Values(url, metadata,
+                Status.FETCHED));
+
+        this.collector.ack(tuple);
+    }
+
+    private List<Values> parseSiteMap(String url, byte[] content,
+            String contentType) {
+
+        crawlercommons.sitemaps.SiteMapParser parser = new crawlercommons.sitemaps.SiteMapParser(
+                strictMode);
+
+        AbstractSiteMap siteMap = null;
+        try {
+            siteMap = parser.parseSiteMap(contentType, content, new URL(url));
+        } catch (Exception e) {
+            LOG.error("Exception while parsing sitemap", e);
+            return null;
+        }
+
+        List<Values> links = new ArrayList<Values>();
+
+        if (siteMap.isIndex()) {
+            SiteMapIndex smi = ((SiteMapIndex) siteMap);
+            Collection<AbstractSiteMap> subsitemaps = smi.getSitemaps();
+            // keep the subsitemaps as outlinks
+            // they will be fetched and parsed in the following steps
+            Iterator<AbstractSiteMap> iter = subsitemaps.iterator();
+            while (iter.hasNext()) {
+                String s = iter.next().getUrl().toExternalForm();
+                // TODO apply filtering to outlinks
+                // TODO configure which metadata gets inherited from parent
+                HashMap<String, String[]> metadata = KeyValues.newInstance();
+                KeyValues.setValue(isSitemapKey, metadata, "true");
+                Values ol = new Values(s, metadata, Status.DISCOVERED);
+                links.add(ol);
+                LOG.debug(url + " : [sitemap] " + s);
+            }
+        }
+        // sitemap files
+        else {
+            SiteMap sm = ((SiteMap) siteMap);
+            // TODO see what we can do with the LastModified info
+            Collection<SiteMapURL> sitemapURLs = sm.getSiteMapUrls();
+            Iterator<SiteMapURL> iter = sitemapURLs.iterator();
+            while (iter.hasNext()) {
+                SiteMapURL smurl = iter.next();
+                double priority = smurl.getPriority();
+                // TODO handle priority in metadata
+                ChangeFrequency freq = smurl.getChangeFrequency();
+                // TODO convert the frequency into a numerical value and handle
+                // it in metadata
+                // TODO configure which metadata gets inherited from parent
+                String s = smurl.getUrl().toExternalForm();
+                // TODO apply filtering to outlinks
+                HashMap<String, String[]> metadata = KeyValues.newInstance();
+                KeyValues.setValue(isSitemapKey, metadata, "false");
+                Values ol = new Values(s, metadata, Status.DISCOVERED);
+                links.add(ol);
+                LOG.debug(url + " : [sitemap] " + s);
```

Comment:
good idea. I discovered this only recently and keep forgetting to use it. I will open a new issue to do the same everywhere else in the project